contracts/sdk/indexer/db.ts

```diff
@@ -284,30 +336,139 @@ export class NFTDatabaseConnection {
     return leafIdxs;
   }
 
-  async getProof(hash: Buffer, treeId: string, check: boolean = true): Promise<Proof | null> {
+  async getMaxSeq(treeId: string) {
+    let res = await this.connection.get(
+      `
+        SELECT max(seq) as seq
+        FROM merkle 
+        WHERE tree_id = ?
+      `,
+      treeId
+    );
+    if (res) {
+      return res.seq;
+    } else {
+      return null;
+    }
+  }
+  async getInferredProof(
```

Comment:
add comment:
// if a gap is detected, will infer a proof over the gap, otherwise will return a normal proof